app/uk/gov/hmrc/icedsubscriptionfrontend/services/AuthService.scala

```diff
@@ -46,16 +46,18 @@ class AuthService @Inject()(val authConnector: AuthConnector)(implicit ec: Execu
   // the exception that is thrown in a particular scenario...
   def authenticate()(implicit hc: HeaderCarrier): Future[AuthResult] =
     authorised(AuthProviders(AuthProvider.GovernmentGateway, AuthProvider.Verify))
-      .retrieve(allEnrolments and affinityGroup and credentials) { retrievals =>
+      .retrieve(allEnrolments and credentialRole and affinityGroup and credentials) { retrievals =>
         import UserType._
 
         val userType = retrievals match {
-          case _ ~ _ ~ Some(Credentials(_, VerifyProviderType))                                    => UnsupportedVerifyUser
-          case enrolments ~ Some(AffinityGroup.Organisation) ~ _ if hasActiveEnrolment(enrolments) => AlreadyEnrolled
-          case _ ~ Some(AffinityGroup.Organisation) ~ _                                            => NotEnrolled
-          case _ ~ Some(AffinityGroup.Individual) ~ _                                              => UnsupportedAffinityIndividual
-          case _ ~ Some(AffinityGroup.Agent) ~ _                                                   => UnsupportedAffinityAgent
-          case _                                                                                   => NonGovernmentGatewayUser
+          case _ ~ _ ~ Some(Credentials(_, VerifyProviderType)) => UnsupportedVerifyUser
+          case enrolments ~ Some(User) ~ Some(AffinityGroup.Organisation) ~ _ if hasActiveEnrolment(enrolments) =>
```

Comment:
Get cyclomatic complexity warning. Maybe replace all of the cases that match on `Some(AffinityGroup.Organisation)` with a single case something that makes it clear that all possibilities are being catered for:
```
          case enrolments ~ role ~ Some(AffinityGroup.Organisation) ~ _ =>
            if (isAdmin(role)) {
              if (hasActiveEnrolment(enrolments)) AlreadyEnrolled else NotEnrolled
            } else {
              WrongCredentialRole
            }
```